<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Phylogenetics</title>

		<meta name="author" content="Trevor Bedford">

		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<link rel="stylesheet" href="../reveal/css/reveal.min.css">
		<link rel="stylesheet" href="../reveal/css/theme/trvrb.css" id="theme">
		<link rel="stylesheet" href="../reveal/css/font-awesome/css/font-awesome.min.css">

		<!-- For syntax highlighting -->
		<link rel="stylesheet" href="../reveal/lib/css/zenburn.css">

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div style="position: absolute; top:10px; left:10px; z-index:100;">
			<a href="http://bedford.io/projects/sismid/sequences/">
				<i class="fa fa-times-circle" style="color: #bbb; opacity: 0.1;"></i>
			</a>
		</div>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
			<div class="slides">

				<section data-background="#ACD1A0">
					<h2 class="title">Phylogenetics</h2>
				</section>

				<section data-background="#000000">
					<img class="stretch" src="images/darwin_phylogeny.jpg">
				</section>

				<section data-background="#000000">
					<img class="stretch" src="images/darwin_phylogeny_tattoo.jpg">
				</section>

				<section data-transition="none">
					<h3>Phylogeny describes evolutionary relationships</h3>
					<img class="stretch" src="images/parsimony_1.png">
				</section>

				<section data-transition="none">
					<h3>Phylogeny is usually a hypothesis based on characteristics of sampled taxa</h3>
					<img class="stretch" src="images/parsimony_2.png">
				</section>

				<section data-transition="none">
					<h3>Phylogeny implies a series of mutational events leading to observed tip states</h3>
					<img class="stretch" src="images/parsimony_3.png">
				</section>

				<section>
					<h3>
					<a href="https://en.wikipedia.org/wiki/Maximum_parsimony_(phylogenetics)">Parsimony</a> is
					based on <a href="https://en.wikipedia.org/wiki/Occam's_razor">Occam's razor</a></h3>
					<p>
					Among competing
					hypotheses that predict equally well, the one with the fewest assumptions should be selected.</p>
				</section>

				<section>
					<h3>Parsimony suggests this topology requires 3 mutations at minimum</h3>
					<img class="stretch" src="images/parsimony_4.png">
				</section>

				<section>
					<h3>Parsimony suggests both topologies equally tenable</h3>
					<img class="stretch" src="images/parsimony_5.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more likely under parsimony?</i></h3>
					<img class="stretch" src="images/parsimony_exercise_1.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more likely under parsimony?</i></h3>
					<img class="stretch" src="images/parsimony_exercise_2.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more likely under parsimony?</i></h3>
					<img class="stretch" src="images/parsimony_exercise_3.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more likely under parsimony?</i></h3>
					<img class="stretch" src="images/parsimony_exercise_4.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more likely under parsimony?</i></h3>
					<img class="stretch" src="images/parsimony_exercise_5.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more likely under parsimony?</i></h3>
					<img class="stretch" src="images/parsimony_exercise_6.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more likely under parsimony?</i></h3>
					<img class="stretch" src="images/parsimony_exercise_7.png">
				</section>

				<section data-transition="none">
					<h3><i>Exercise: which topology is more likely under parsimony?</i></h3>
					<img class="stretch" src="images/parsimony_exercise_8.png">
				</section>

				<section data-background="#B9D395">
					<h2 class="title">Phylogenetic inference</h2>
				</section>

				<section>
					<img class="stretch" src="images/phylogenetic_inference.png">
				</section>

				<section>
					<h3>"Data" is generally a sequence alignment</h3>
					<img class="stretch" src="images/phylogenetic_inference_alignment.png">
				</section>

				<section>
					<h3>Phylogeny structures site patterns</h3>
					<img class="stretch" src="images/phylogenetic_inference_site_patterns.png">
					<div class="citation">
							Buffalo. 2015
					</div>
				</section>

				<section style="text-align: left;">
					<h3>Tree space is vast</h3>
					<p>
					There are (2<i>n</i>-3)!! rooted trees for <i>n</i> taxa
					<ul>
						<li>3 taxa: 3 trees</li>
						<li>5 taxa: 105 trees</li>
						<li>10 taxa: 34,459,425 trees</li>
						<li>20 taxa: 8.2 &times; 10<sup>21</sup> trees</li>
						<li>50 taxa: 2.8 &times; 10<sup>76</sup> trees</li>
						<li>100 taxa: 3.3 &times; 10<sup>184</sup> trees</li>
					</ul>
					</p>
				</section>

				<section>
					<h3>Solution space is rugged</h3>
					<img class="stretch" src="images/phylogenetic_inference_solution_space.png">
				</section>

				<section style="text-align: left;">
					<h3>Types of phylogenetic inference methods</h3>
					<p>
					<ul>
						<li>Distance-based (neighbor-joining, fast, heuristic)</li>
						<li>Parsimony (fast, "model-free")</li>
						<li>Maximum likelihood (infers model of mutation, accurate, examples: FastTree, RAxML, IQTREE)</li>
						<li>Bayesian (like ML, but requires prior, produces estimates of uncertainty, examples: MrBayes, BEAST)</li>
					</ul>
					</p>
				</section>

				<section>
					<h3>Inference is a tree topology, branch lengths and ancestral states</h3>
					<img class="stretch" src="images/ancestral_state_tree.png">
					<div class="citation">
						<a href="http://nextflu.org/">nextflu.org</a>
					</div>
				</section>

				<section data-background="#56A0AE">
					<h2 class="title">Molecular clocks and dated phylogenies</h2>
				</section>

				<section>
					<h3>Mutations tend to accumulate in a clock-like fashion</h3>
					<img class="stretch" src="images/clock_ncov_tree.png">
				</section>

				<section>
					<h3>"Root-to-tip" plots show temporal signal</h3>
					<img class="stretch" src="images/clock_ncov_root_to_tip.png">
				</section>

				<section>
					<h3>Allows conversion between branch length and time</h3>
					<img class="stretch" src="images/clock_ncov_timetree.png">
				</section>

				<section>
					<h3>Dated phylogenies provide real-world context</h3>
					<img class="stretch" src="images/clock_ncov_introductions.png">
				</section>

				<section data-background="#63AC9A">
					<h2 class="title">Inference of discrete traits</h2>
				</section>

				<section>
					<h3>"Data" is a phylogeny and tip states</h3>
					<p>States include nucleotides, amino acids, geo locations, hosts, etc...</p>
					<img class="stretch" src="images/phylogeo_inference_data.png">
				</section>

				<section>
					<h3>Model infers transition matrix and ancestral states</h3>
					<img class="stretch" src="images/phylogeo_inference_model.png">
				</section>

				<section>
					<h3>Rare transitions, short branches and many taxa increase confidence</h3>
					<img class="stretch" src="images/phylogeo_inference_rare_transitions.png">
				</section>

				<section data-background="#71B486">
					<h2 class="title">Phylogeography</h2>
				</section>

				<section>
					<h3>Nesting patterns are informative</h3>
					<img class="stretch" src="images/phylogeo_inference_nesting.png">
				</section>

				<section>
					<h3>Zika phylogeny infers an origin in northeast Brazil</h3>
					<img class="stretch" src="images/phylogeo_zika_tree.png">
					<div class="citation">
						<a href="http://bedford.io/papers/faria-zika-in-brazil/">Faria et al. 2017. Nature.</a>
					</div>
				</section>

				<section>
					<h3>Influenza phylogeny shows repeated spread from E-SE Asia</h3>
					<img class="stretch" src="images/phylogeo_flu_tree.png">
					<div class="citation">
						<a href="http://bedford.io/papers/bedford-global-circulation/">
							Bedford et al. 2015. Nature.
						</a>
					</div>
				</section>

				<section>
					<h3>Influenza transitions mirror air travel network</h3>
					<img class="stretch" src="images/phylogeo_flu_map.png">
					<div class="citation">
						<a href="http://bedford.io/papers/bedford-global-circulation/">
							Bedford et al. 2015. Nature.
						</a>
					</div>
				</section>

				<section>
					<h3>Detailed inference of SARS-CoV-2 <br> introductions into the UK</h3>
					<div class="pull-right" style="vertical-align: bottom; display: table-cell;">
						<img width="42%" src="images/phylogeo_ncov_uk_lineages_examples.png">
						&nbsp;
						<img width="54%" src="images/phylogeo_ncov_uk_lineages_sources.png">
					</div>
					<div class="citation">
						<a href="https://science.sciencemag.org/content/371/6530/708">du Plessis et al. 2021. Science.</a>
					</div>
				</section>

				<section data-background="#81BA72">
					<h2 class="title">Reservoir species and host jumps</h2>
				</section>

				<section>
					<h3>Influenza has a reservoir in wild birds, spillovers sometimes become endemic</h3>
					<img class="stretch" src="images/spillover_influenza_tree.png">
				</section>

				<section>
					<h3>Each Ebola outbreak derives from a separate spillover event</h3>
					<img class="stretch" src="images/spillover_ebola_tree.png">
					<div class="citation">
						<a href="http://science.sciencemag.org/content/345/6202/1369">
							Gire et al. 2014. Science.
						</a>
					</div>
				</section>

				<section>
					<h3>MERS-CoV has frequent spillover events, but limited human-to-human transmission</h3>
					<img class="stretch" src="images/spillover_mers_tree.png">
					<div class="citation">
						<a href="http://bedford.io/papers/dudas-mers-structure/">
							Dudas et al. 2017. bioRxiv.
						</a>
					</div>
				</section>

				<section data-background="#94BD62">
					<h2 class="title">Reassortment and recombination</h2>
				</section>

				<section>
					<h3>Influenza B reassorts across segments</h3>
					<img class="stretch" src="images/recombination_flu_b_trees.png">
					<div class="citation">
						<a href="http://bedford.io/papers/dudas-flub-reassortment/">Dudas et al. 2015. Mol Biol Evol.</a>
					</div>
				</section>

				<section>
					<h3>Reassortment splits gene constellations</h3>
					<img class="stretch" src="images/recombination_flu_b_railway.png">
					<div class="citation">
						<a href="http://bedford.io/papers/dudas-flub-reassortment/">Dudas et al. 2015. Mol Biol Evol.</a>
					</div>
				</section>

				<section>
					<h3>Recombination in seasonal coronaviruses</h3>
					<img class="stretch" src="images/recombination_hcov.png">
					<div class="citation">
						<a href="https://bedford.io/papers/muller-cov-recombination/">MÃ¼ller et al. 2021. bioRxiv.</a>
					</div>
				</section>

				<!-- <section>
					<h3>Abundant recombination during within-host HIV evolution</h3>
					<img class="stretch" src="images/recombination_hiv_trees.png">
					<div class="citation">
						<a href="https://elifesciences.org/articles/11282">Zanini et al. 2015. eLife.</a>
					</div>
				</section>

				<section>
					<h3>Linkage disequilibrium decays with physical distance</h3>
					<img class="stretch" src="images/recombination_hiv_ld.png">
					<div class="citation">
						<a href="https://elifesciences.org/articles/11282">Zanini et al. 2015. eLife.</a>
					</div>
				</section> -->

				<section style="text-align: left;">
					<h3>Summary</h3>
					<p>
					<ul>
						<li>Phylogenetics uses sequence data to infer trees along with ancestral states</li>
						<li>Molecular clocks provide a real-world context to evolutionary events</li>
						<li>Phylogeny reveals behavior of discrete traits, like geographic location and host state</li>
					</ul>
					</p>
				</section>


			</div>

		</div>

		<script src="../reveal/lib/js/head.min.js"></script>
		<script src="../reveal/js/reveal.min.js"></script>
		<script src="../reveal/js/config.js"></script>

	</body>
</html>
