<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Forecasting</title>

		<meta name="author" content="Sarah Cobey">

		<meta name="apple-mobile-web-app-capable" content="yes" />
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<link rel="stylesheet" href="../reveal/css/reveal.min.css">
		<link rel="stylesheet" href="../reveal/css/theme/trvrb.css" id="theme">
		<link rel="stylesheet" href="../reveal/css/font-awesome/css/font-awesome.min.css">

		<!-- For syntax highlighting -->
		<link rel="stylesheet" href="../reveal/lib/css/zenburn.css">

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div style="position: absolute; top:10px; left:10px; z-index:100;">
			<a href="/sismid/forecasting">
				<i class="fa fa-times-circle" style="color: #bbb; opacity: 0.1;"></i>
			</a>
		</div>

		<div class="reveal">

			<!-- Any section element inside of this container is displayed as a slide -->
			<div class="slides">

				<section data-background="#CC3333">
					<h2 class="title">Forecasting</h2>
				</section>

				<section>
					<h3>Forecasting is a big problem</h3>
					<img class="stretch" src="images/globe.png">
				</section>

                <section>
                    <h3>Rapid progress</h3>
                    <p>Actual v. predicted distances of Atlantic cyclones
                    <img class="stretch" src="images/cyclone_error.jpg">
                        <div class="citation">
                            <a href="http://www.sciencedirect.com.proxy.uchicago.edu/science/article/pii/S0169534716301185">
                                Gandon et al. 2016
                            </a>
                        </div>
                </section>

				<section>
					<h3>General, overlapping approaches</h3>
					<p>Machine learning
					<p>Statistical models
					<p>Nonlinear forecasting
					<p>"Mechanistic" modeling
				</section>

				<section>
					<h3>DREAM challenges</h3>
					<p>Inference of gene regulatory networks
					<p>from knockout, observational, and synthetic data
					<img src="images/dream.png">
					<div class="citation">
						<a href="http://bioinformatics.oxfordjournals.org/content/early/2012/03/29/bioinformatics.bts143.full.pdf">
							Kuffner et al. 2012
						</a>
					</div>
					<p>Compete Lasso, random forests, Bayesian networks, mutual information, ANOVA, etc.
				</section>

				<section>
					<h3>Ecological niche modeling</h3>
					<img src="images/channel.png">
						<div class="citation">
							<a href="http://www.nature.com/nmeth/journal/v9/n6/full/nmeth.1975.html">
								Larsen et al. 2012
							</a>
						</div>
						<p>An artificial neural network that included microbial interactions performed best.
				</section>

				<section>
					<h3>Nonlinear forecasting</h3>
					<p>Reconstruct attractor ("library") from time series
					<p>Use attractor to make short-term predictions</p>
					<div class="citation">
						<a href="http://www.nature.com/nature/journal/v344/n6268/abs/344734a0.html">
							Sugihara and May 1990
						</a>
					</div>
				</section>

								<section>
					<h3>Forecasting communities</h3>
					<img class="stretch" src="images/community.jpg">
						<div class="citation">
							<a href="http://rspb.royalsocietypublishing.org/content/283/1822/20152258">Deyle et al. 2015</a>
						</div>
				</section>

				<section>
					<h3>Dynamics may be chaotic</h3>
					<img class="stretch" src="images/lorenz.png">
				</section>

				<section>
					<h3>Predictions with chaos: short shelf life</h3>
					<p>Trajectories in chaotic attractors diverge
					<p> $$ \lvert \delta \textbf{Z}(t)\rvert \approx e^{\lambda t}\lvert\delta\textbf{Z}_0\rvert$$
					<p> $\lambda$ is the Lyapunov exponent
					<p> (so with chaos, $\lambda>0$)
				</section>

				<section>
					<section>
					<h3>How to forecast</h3>
					<ul>
						<li>Choose an embedding dimension $E$ and lag $\tau$</li>
						<li>Each point in $E$-dimensional space: $\{x_t,x_{t-\tau},x_{t-2\tau},...,x_{t-(E-1)\tau}\}$</li>
						<li>Construct these points from the time series</li>
						<li>Define a point to predict ("predictee")</li>
						<li>See where predictee's $E+1$ nearest neighbors wind up $t$ steps into the future</li>
						<li>Measure correlations $\rho$ between predictee's observed future state and neighbors' weighted predictions</li>
					</ul>
					</section>

					<section>
					<h3>Choosing $E$ and $\tau$</h3>
					<p>An unsolved problem
					<p>Use $E$ (and $\tau$) that yield best predictions</p>
					<img class="stretch" src="images/E_selection.png">
						<div class="citation">
							<a href="http://www.nature.com/nature/journal/v344/n6268/abs/344734a0.html">Sugihara and May 1990</a>
						</div>
						</section>
				</section>



				<section>
					<h3>Observational noise v. chaos</h3>
					<img class="stretch" src="images/chaos_noise.png">
						<div class="citation">
							<a href="http://www.nature.com/nature/journal/v344/n6268/abs/344734a0.html">Sugihara and May 1990</a>
						</div>
				</section>

				<section>
					<h3>Predicting flu</h3>
					<img class="stretch" src="images/subtypes_season.png">
						<div class="citation">
							<a href="http://journals.plos.org/plosmedicine/article?id=10.1371/journal.pmed.1001051">Goldstein et al. 2011</a>
						</div>
				</section>

				<section>
					<h3>Epidemic sizes negatively correlated</h3>
					<img class="stretch" src="images/subtypes_correlations.png">
						<div class="citation">
							<a href="http://journals.plos.org/plosmedicine/article?id=10.1371/journal.pmed.1001051">Goldstein et al. 2011</a>
						</div>
				</section>

				<section>
					<h3>Predict based on cumulative incidence</h3>
					<img src="images/cip_equations.png" style="width: 50%; height: 50%;">
                        <p>where $I(s)$ is the incidence in week $s$, $h$ is a strain-specific incidence threshold, $T$ is the time of crossing $h$, and $Y$ is the strain's whole-season cumulative incidence proxy.
						<div class="citation">
							<a href="http://journals.plos.org/plosmedicine/article?id=10.1371/journal.pmed.1001051">Goldstein et al. 2011</a>
						</div>
					</section>

				<section>
					<h3>Predictions v. observations (H3N2)</h3>
					<img class="stretch" src="images/h3n2_prediction.png">
						<div class="citation">
							<a href="http://journals.plos.org/plosmedicine/article?id=10.1371/journal.pmed.1001051">Goldstein et al. 2011</a>
						</div>
				</section>


				<section>
					<h3>Predicting flu in Hong King</h3>
					<p>Aim: Predict peak timing and magnitude
					<img src="images/hongkong.png" style="width: 50%; height: 50%;">
						<div class="citation">
							<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004383">
								Yang et al. 2015
							</a>
						</div>
				</section>

				<section>
					<h3>Mechanistic model and particle filter</h3>
					<img src="images/SI_equations.png">
						<div class="citation">
							<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004383">
								Yang et al. 2015
							</a>
						</div>
						<p>Achieves 37% accuracy with 1-3 week lead, ~50% at 0 week lead
					</section>


				<section>
                    <h3><a href="https://predict.cdc.gov/">Epidemic Prediction Initiative</a>
				</section>

				<section>
                    <h3>Forecasting SARS-CoV-2</h3>
						<img class="stretch" src="images/birx_ihme.jpg">
						<div class="citation">
							<a href="https://www.cnn.com/2020/04/08/politics/deborah-birx-social-distancing-models/index.html">
								CNN
							</a>
						</div>
				</section>

				<section>
                    <h3>Unusual error</h3>
						<img class="stretch" src="images/ihme_error.png">
				</section>

				<section>
                    <h3>A statistical model</h3>
						<img class="stretch" src="images/ihme_eqns.png">
				</section>

				<section>
                    <h3>Some precedent: Farr's Law</h3>
						<img class="stretch" src="images/farrs_law_jama.png">
				</section>

				<section>
                    <h3><a href="https://covid19forecasthub.org/">(U.S.) COVID-19 Forecast Hub</a>
						<div class="citation">
							see <a href="https://www.pnas.org/doi/10.1073/pnas.2113561119">
								Cramer et al. 2022
							</a>
						</div>
				</section>

				<section>
					<h3>Comparing models</h3>
					<img class="stretch" src="images/model_table.png">
						<div class="citation">
							<a href="https://www.pnas.org/doi/10.1073/pnas.2113561119">
								Cramer et al. 2022
							</a>
						</div>
				</section>

				<section>
					<h3>Evaluating performance</h3>
					<img class="stretch" src="images/model_rank.png">
						<div class="citation">
							<a href="https://www.pnas.org/doi/10.1073/pnas.2113561119">
								Cramer et al. 2022
							</a>
						</div>
				</section>

				<section>
					<h3>Failure to predict change</h3>
					<img class="stretch" src="images/COVID_ensemble_forecasts.png" style="width: 50%; height: 50%;">
						<div class="citation">
							<a href="https://delphi.cmu.edu/blog/2021/09/30/on-the-predictability-of-covid-19/">
								Reich et al. 2021
							</a>
						</div>
				</section>

				<section data-transition="fade" data-background="#000000">
					<h1 class="title">Evolutionary forecasting</h1>
				</section>

				<section>
					<h3>Influenza vaccine strain selection strategy</h3>
					<p class="smaller">
						General strategy for antigenically evolving seasonal influenza viruses is attempt to match vaccine strain as close as possible to circulating viruses
					</p>
					<img class="stretch" src="images/flu_h3n2_tree_vaccines_2014_2022.png">
					<div class="citation">
						<a href="https://nextstrain.org/flu/seasonal/h3n2/ha/6y">nextstrain.org</a>
					</div>
				</section>

				<section>
					<h3>Vaccine strain selection timeline</h3>
					<p class="smaller">
						Due to manufacturing and distribution, vaccine strain selection occurs in Feb for an fall vaccination campaign ahead of seasonal influenza epidemic
					</p>
					<img class="stretch" src="images/flu_vaccine_schedule.jpg">
				</section>

				<section>
					<h3>Fitness models project strain frequencies</h3>
					<p class="smaller">Future frequency $x_i(t+\Delta t)$ of strain $i$ derives from strain fitness $f_i$ and present day frequency $x_i(t)$, such that</p>
					<p class="smaller">$$x_i(t+\Delta t) = \frac{1}{Z(t)} \, x_i(t) \, \mathrm{exp}(f_i \, \Delta t)$$</p>
					<p class="smaller">
						Strain frequencies at each timepoint are normalized by total frequency $Z(t)$.
						This captures clonal interference between competing lineages.
					</p>
					<img class="stretch" src="images/clonal_interferance.jpg">
					<div class="citation">
						<a href="http://www.nature.com/nature/journal/v507/n7490/full/nature13087.html">&#321;uksza and L&auml;ssig. 2014</a>
					</div>
				</section>

				<section>
					<h3>Match strain forecast to retrospective circulation</h3>
					<img class="stretch" src="images/prediction_fitness_model_schematic.png">
					<div class="citation">
						<a href="https://bedford.io/papers/huddleston-flu-forecasting/">Huddleston et al. 2020</a>
					</div>
				</section>

				<section>
					<h3>Strain fitness estimated from viral attributes</h3>
					<p class="smaller">The fitness $f$ of strain $i$ is estimated as</p>
					<p class="smaller">$$f_i = \beta^\mathrm{A} \, f_i^\mathrm{A} + \beta^\mathrm{B} \, f_i^\mathrm{B} + \ldots$$</p>
					<p class="smaller">where $f^A$, $f^B$, etc... are different standardized viral attributes and
						$\beta^A$, $\beta^B$, etc... coefficients are trained based on historical evolution</p>
					<br>
					<table class="smaller">
						<thead>
							<tr>
								<th width="28%">Antigenic drift</th>
								<th width="38%">Intrinsic fitness</th>
								<th width="33%">Recent growth</th>
							</tr>
						</thead>
						<tbody>
							<tr>
								<td>epitope mutations</td>
								<td>non-epitope mutations</td>
								<td>local branching index</td>
							</tr>
							<tr>
								<td>HI titers</td>
								<td>DMS data (via Bloom lab)</td>
								<td>delta frequency</td>
							</tr>
						</tbody>
					</table>
				</section>

				<section>
					<h3>Model successfully predicts clade growth and best pick from model is generally close to best possible retrospective pick</h3>
					<img class="stretch" src="images/prediction_validation_clade_growth_ranking.png">
					<div class="citation">
						<a href="https://bedford.io/papers/huddleston-flu-forecasting/">Huddleston et al. 2020. eLife.</a>
					</div>
				</section>

				<section class="left-align">
					<h3>Main issue</h3>
					<p>
						Strain fitness $f_i$ is largely fixed by the "fundamentals" of the strain rather than being learned from frequency behavior.
					</p>
				</section>

				<section>
					<h3>Genetic relationships of globally sampled SARS-CoV-2 to present</h3>
					<img class="stretch" src="images/ncov_variants_tree_unrooted.png">
					<div class="citation">
						<a href="https://nextstrain.org/ncov/gisaid/global?l=unrooted&m=div">nextstrain.org</a>
					</div>
				</section>

				<section>
					<h3>Rapid displacement of existing diversity by emerging variants</h3>
					<img class="stretch" src="images/ncov_tree_freq.png">
					<div class="citation">
						<a href="https://nextstrain.org/ncov/gisaid/global">nextstrain.org</a>
					</div>
				</section>

				<section>
					<h3>Population genetic expectation of variant frequency under selection</h3>
					<p class="smaller">$x' = \frac{x \, (1+s)}{x \, (1+s) + (1-x)}$ for frequency $x$ in one generation with selective advantage $s$</p>
					<p class="smaller">$x(t) = \frac{x_0 \, (1+s)^t}{x_0 \, (1+s)^t + (1-x_0)}$ for initial frequency $x_0$ over $t$ generations</p>
					<p class="smaller">Trajectories are linear once logit transformed via $\mathrm{log}(\frac{x}{1 - x})$</p>
					<img class="stretch" src="images/ncov_variants_selection_logit_trajectories.png">
				</section>

				<section>
					<h3>Variants show consistent frequency dynamics in logit space</h3>
					<img class="stretch" src="images/ncov_variants_omicron_countries_frequencies_logit_2022_04_08.png">
				</section>

				<!-- <section>
					<h3>Variants show consistent frequency dynamics in logit space</h3>
					<img class="stretch" src="images/ncov_variants_delta_countries_frequencies_logit_2021_09_07.png">
				</section> -->

				<section>
					<h3>Multinomial logistic regression</h3>
					<p class="smaller">
						Multinomial logistic regression models the probability of a virus sampled at time $t$ belonging to variant $i$ as
					</p>
					<p class="smaller">
						$$\mathrm{Pr}(X = i) = x_i(t) = \frac{p_i \, \mathrm{exp}(f_i \, t)}{\sum_{1 \le j \le n} p_j \, \mathrm{exp}(f_j \, t) }$$
					</p>
					<p class="smaller">
						where the model has $2n$ parameters consisting of $p_i$ the frequency of variant $i$ at initial timepoint and $f_i$ the
						growth rate or fitness of variant $i$ for $n$ variants.
					</p>
					<p class="smaller">
						The model is fit to minimize "log loss" of predicted variant vs observed variant across observations in dataset.
					</p>
				</section>

				<section>
					<h3>Multinomial logistic regression fits variant frequencies well</h3>
					<img class="stretch" src="images/sarscov2_variant_rt_logistic_regression_2021_09_01.png">
				</section>

				<section>
					<h3>Original VOC viruses had substantially increased transmissibility</h3>
					<img class="stretch" src="images/sarscov2_variant_rt_growth_advantage.png">
					<div class="citation">
						Model from <a href="https://bedford.io/papers/figgins-rt-from-frequency-dynamics/">Figgins and Bedford. 2022. medRxiv.</a>
					</div>
				</section>

				<section>
					<h3>Clade and lineage forecasts continuously updated</h3>
					<img class="stretch" src="images/ncov_forecasts_clades.png">
					<div class="citation">
						<a href="https://nextstrain.org/sars-cov-2/forecasts/">Figgins, Lee, Hadfield. nextstrain.org</a>
					</div>
				</section>

				<section>
					<p>Multinomial logistic regression should work well for SARS-CoV-2 prediction, except new variants have been emerging
						fast enough that the prediction horizon is really quite short</p>
				</section>

				<section data-background="#99CCFF">
					<h3>What other models would you test?</h3>
				</section>

				<section data-background="#99CCFF">
					<h3>What limits prediction with other pathogens?</h3>
				</section>

		</div>

		<script src="../reveal/lib/js/head.min.js"></script>
		<script src="../reveal/js/reveal.min.js"></script>
		<script src="../reveal/js/config.js"></script>

	</body>
</html>
